charmcraft/application/main.py

```diff
@@ -31,7 +31,8 @@
 APP_METADATA = AppMetadata(
     name="charmcraft",
     summary=GENERAL_SUMMARY,
-    ProjectClass=models.charmcraft.Project,
+    # This will change soon so just ignore it for now.
+    ProjectClass=models.charmcraft.Project,  # type: ignore[arg-type]
```

Comment:
Interesting, how is it going to look like?

Reply:
In #1328, charmcraft/models/project.py